tools/validation-generator/src/main/java/io/spine/tools/validate/MessageValidatorFactory.java

```diff
@@ -0,0 +1,237 @@
+import static javax.lang.model.element.Modifier.STATIC;
+
+/**
+ * A factory of message validation code.
+ */
+public final class MessageValidatorFactory {
+
+    private static final String VALIDATE_METHOD = "validate";
+    private static final String MESSAGE_PARAMETER = "msg";
+    private static final String MESSAGE_VARIABLE = "msg";
+    private static final String RETURN_N = "return $N";
+    private static final String VIOLATIONS = "violations";
+
+    @SuppressWarnings("UnstableApiUsage")
+    private static final Type listOfViolations =
+            new TypeToken<List<ConstraintViolation>>() {}.getType();
+    @SuppressWarnings("UnstableApiUsage")
+    private static final Type arrayListOfViolations =
+            new TypeToken<ArrayList<ConstraintViolation>>() {}.getType();
+
+    private final MessageType type;
+    private final NestedClassName messageSimpleName;
+    private final String validatorSimpleName;
+
+    /**
+     * Creates a new {@code MessageValidatorFactory} for the given type.
+     *
+     * @param type
+     *         type of the message to validate
+     */
+    public MessageValidatorFactory(MessageType type) {
+        this.type = type;
+        this.messageSimpleName = NestedClassName.from(type.javaClassName());
+        this.validatorSimpleName = nameForValidator(type);
+    }
+
+    private static String nameForValidator(MessageType type) {
+        StringBuilder candidate = new StringBuilder("Validator");
+        while (nameClashes(type, candidate.toString())) {
+            candidate.append('$');
+        }
+        return candidate.toString();
+    }
+
+    private static boolean nameClashes(MessageType type, String name) {
+        boolean withOwnName = type.simpleJavaClassName()
+                                  .value()
+                                  .equals(name);
+        if (withOwnName) {
+            return true;
+        }
+        boolean withNestedName = type
+                .nestedDeclarations()
+                .stream()
+                .anyMatch(nested -> nested.simpleJavaClassName()
+                                          .value()
+                                          .equals(name));
+        return withNestedName;
+    }
+
+    /**
+     * Generates the {@code Validator} class for the associated message type.
+     *
+     * <p>The {@code Validator} class is supposed to be nested inside the message class. It is
+     * declared {@code private} and {@code static}.
+     *
+     * <p>The name of the class is most often {@code Validator}. However, when a name clash occurs,
+     * the name is padded with the {@code $} (dollar sign). For instance if the message itself is
+     * called {@code Validator}, the validator class will be called {@code Validator$}. Since
+     * the class is {@code private}, this naming is not exposed to the outer scope.
+     *
+     * <p>The only method of the class is:
+     * <pre>
+     *     {@code
```

Comment:
Let's kill the `{@code}`. Please refer to one of our annotations (such as `@Command`) where I had experimented with the code samples in Javadoc and come to some conclusions.